packages/benchmarks/src/textToDriver/makeDatabases/loadBraintrustDatasets.ts

```diff
@@ -0,0 +1,108 @@
+import { initDataset } from "braintrust";
+import { EJSON } from "bson";
+import {
+  CollationOptions,
+  CreateIndexesOptions,
+  Document,
+  IndexDirection,
+  IndexSpecification,
+} from "mongodb-rag-core";
+
+interface LoadBraintrustDatasetParams {
+  apiKey: string;
+  projectName: string;
+  datasetName?: string;
+}
+
+export async function loadBraintrustMetadata({
+  apiKey,
+  projectName,
+  datasetName = "database-metadata",
+}: LoadBraintrustDatasetParams) {
+  const dataset = initDataset(projectName, {
+    apiKey,
+    dataset: datasetName,
+  });
+  const metadata: DatabaseMetadata[] =
+    // TODO: why do i have to cast this as unknown?
+    (
+      (await dataset.fetchedData()) as unknown as DatabaseMetadataDatasetEntry[]
+    ).map(
+      (m) =>
+        ({
+          databaseName: m.input.databaseName,
+          collections: m.input.collections.map((c) => ({
+            collectionName: c.collectionName,
+            indexes: c.indexes.map((idx) =>
+              EJSON.deserialize(idx)
+            ) as unknown as CollectionMetadata["indexes"],
```

Comment:
Don't love this cast either, might be solved with zod parsing?

Reply:
i think so. will implement